app/api/search/entitiesIndex.js

```diff
@@ -7,7 +7,70 @@ import { entityDefaultDocument } from 'shared/entityDefaultDocument';
 
 import elastic from './elastic';
 
-const bulkIndex = (docs, _action = 'index', elasticIndex) => {
+function truncateStringToLuceneLimit(str) {
+  const LUCENE_BYTES_LIMIT = 32766;
+  const bytes = Buffer.from(str);
+  return bytes.slice(0, Math.min(LUCENE_BYTES_LIMIT, bytes.length)).toString();
+}
+
+function truncateLongFieldValue(fieldPath, docData) {
+  fieldPath.reduce((path, prop) => {
+    const currentPath = path;
+    if (currentPath[prop] !== undefined && currentPath[prop] !== Object(currentPath[prop])) {
+      currentPath[prop] = truncateStringToLuceneLimit(currentPath[prop]);
+    } else if (Array.isArray(currentPath) && currentPath[0][prop] !== Object(currentPath[prop])) {
+      currentPath[0][prop] = truncateStringToLuceneLimit(currentPath[0][prop]);
+    }
+    return currentPath[prop];
+  }, docData);
+}
+
+const handledFailedDocsByLargeFieldErrors = async (body, errors) => {
+  if (errors.length === 0) return '';
+  const invalidFields = new Set();
+  errors.forEach(error => {
+    const docIndex = body.findIndex(
+      doc => doc.index !== undefined && doc.index._id === error.index._id
+    );
+    const errorExpression = /Document contains at least one immense term in field="(\S+)".+/g;
+    const matches = errorExpression.exec(error.index.error.reason);
+    const docData = body[docIndex + 1];
+    const field = matches[1].split(/\.(?=[^.]+$)/)[0];
+    const fieldPath = field.split('.');
+    truncateLongFieldValue(fieldPath, docData);
+    invalidFields.add(field);
+  });
+  await elastic.bulk({ body, requestTimeout: 40000 });
+  return `max_bytes_length_exceeded_exception. Invalid Fields: ${Array.from(invalidFields).join(
+    ', '
+  )}`;
+};
```

Comment:
@mfacar is this still needed even with the validation we added for the fields ?

Reply:
I thought the same, with the validation the code becomes unreachable, however, in my tests it turned out that this control throws an exception when the title of the entity is bigger too. 
I think it is better to add the validation at title field and delete this control, what do you think?

Reply:
yes i think the same, this is a lot of code for just a specific scenario, validation solves it. 👍

Reply:
unreachable code was removed